File: 1173-34-2364-count-number-of-bad-pairs/1173-34-2364-count-number-of-bad-pairs.js
var countBadPairs = function(nums) {
    let n = nums.length;
    let bads = BigInt(n) * BigInt(n - 1) / 2n; 
    let mp = new Map();

    for (let i = 0; i < n; i++) {
        let key = nums[i] - i;
        let count = mp.get(key) || 0;
        bads -= BigInt(count); 
        mp.set(key, count + 1);
    }

    return Number(bads);
};

File: 1173-34-2364-count-number-of-bad-pairs/1173-34-2364-count-number-of-bad-pairs.java
import java.util.HashMap;
import java.util.Map;

class Solution {
    public long countBadPairs(int[] nums) {
        int n = nums.length;
        long bads = (long) n * (n - 1) / 2;
        Map<Integer, Integer> mp = new HashMap<>();

        for (int i = 0; i < n; i++) {
            int key = nums[i] - i;
            bads -= mp.getOrDefault(key, 0);
            mp.put(key, mp.getOrDefault(key, 0) + 1);
        }

        return bads;
    }
}
